builtin/providers/aws/resource_aws_cloudfront_web_distribution.go

```diff
@@ -0,0 +1,587 @@
+	d.Set("cached_methods", resourceAwsCloudFrontCopyItems(c.DefaultCacheBehavior.AllowedMethods.CachedMethods.Items))
+	d.Set("forwarded_headers", resourceAwsCloudFrontCopyItems(c.DefaultCacheBehavior.ForwardedValues.Headers.Items))
+	d.Set("logging_enabled", c.Logging.Enabled)
+	d.Set("logging_include_cookies", c.Logging.IncludeCookies)
+	d.Set("logging_prefix", c.Logging.Prefix)
+	d.Set("logging_bucket", c.Logging.Bucket)
+	d.Set("aliases", c.Aliases.Items)
+	d.Set("geo_restriction_type", c.Restrictions.GeoRestriction.RestrictionType)
+	d.Set("geo_restrictions", resourceAwsCloudFrontCopyItems(c.Restrictions.GeoRestriction.Items))
+	d.Set("zone_id", "Z2FDTNDATAQYW2")
+
+	d.Set("minimum_ssl", c.ViewerCertificate.MinimumProtocolVersion)
+	d.Set("ssl_support_method", c.ViewerCertificate.SSLSupportMethod)
+	if *c.ViewerCertificate.CloudFrontDefaultCertificate == true {
+		d.Set("certificate_id", "")
+	} else {
+		d.Set("certificate_id", c.ViewerCertificate.IAMCertificateId)
+	}
+
+	// CloudFront distributions supports multiple origins. However most of the above
+	// configuration options also apply to a single origin which would result in
+	// an overwhelming API
+	o := c.Origins.Items[0]
+	d.Set("origin_domain_name", o.DomainName)
+	d.Set("origin_path", o.OriginPath)
+	d.Set("origin_http_port", o.CustomOriginConfig.HTTPPort)
+	d.Set("origin_https_port", o.CustomOriginConfig.HTTPSPort)
+	d.Set("origin_protocol_policy", o.CustomOriginConfig.OriginProtocolPolicy)
+
+	return nil
+}
+
+func resourceAwsCloudFrontWebDistributionUpdate(d *schema.ResourceData, meta interface{}) error {
+	cloudfrontconn := meta.(*AWSClient).cloudfrontconn
+
+	// CloudFront configuration changes requires the ETag of the latest changeset
+	v, err := resourceAwsCloudFrontWebDistributionDistributionRetrieve(d, meta)
+	if err != nil {
+		return err
+	}
+
+	c, err := resourceAwsCloudFrontWebDistributionDistributionConfig(d, meta, v.Distribution.DistributionConfig.CallerReference)
+	if err != nil {
+		return err
+	}
+
+	params := &cloudfront.UpdateDistributionInput{
+		DistributionConfig: c,
+		Id:                 aws.String(string(d.Id())),
+		IfMatch:            v.ETag,
+	}
+
+	_, err = cloudfrontconn.UpdateDistribution(params)
+
+	if err != nil {
+		return fmt.Errorf("Error updating CloudFront distribution: %s", err)
+	}
+
+	err = resourceAwsCloudFrontWebDistributionWaitUntilDeployed(d, meta)
+	if err != nil {
+		return err
+	}
+
+	return resourceAwsCloudFrontWebDistributionRead(d, meta)
+}
+
+func resourceAwsCloudFrontWebDistributionDelete(d *schema.ResourceData, meta interface{}) error {
+	cloudfrontconn := meta.(*AWSClient).cloudfrontconn
+
+	v, err := resourceAwsCloudFrontWebDistributionDistributionRetrieve(d, meta)
+	if err != nil {
+		return err
+	}
+
+	// Do nothing if resource no longer exists
+	if v == nil {
+		return nil
+	}
+
+	// CloudFront distributions must be disabled in order to be deleted
+	if d.Get("enabled") == true {
```

Comment:
You are comparing `interface{}` to `true`: https://github.com/hashicorp/terraform/blob/cf98cbc9adc6df1cabf73ba9689574bbb3905972/helper/schema/resource_data.go#L56
Some casting will be necessary.